UnityProject/Assets/Scripts/Tilemaps/Behaviours/Pathfinding/PressurePathfinder.cs

```diff
@@ -19,71 +19,196 @@ private bool PositionIsOutOfBounds(ChunkedTileMap<MetaDataNode> terrain, Vector3
+		{
+			if (PositionIsOutOfBounds(terrain, start) || PositionIsOutOfBounds(terrain, end))
+				return null;
+
+			var fScore = new Dictionary<Vector3Int, float>();
+			var openSet = new SortedSet<Vector3Int>(new FScoreComparer(fScore));
+			var cameFrom = new Dictionary<Vector3Int, Vector3Int>();
+			var gScore = new Dictionary<Vector3Int, float>();
+			var path = new List<Vector3Int>();
+			Vector3Int lastValidPosition = start;
+
+			openSet.Add(start);
+			gScore[start] = 0;
+			fScore[start] = Heuristic(start, end);
+
+			while (openSet.Count > 0)
+			{
+				var current = openSet.Min;
+				if (current == end)
+				{
+					while (current != start)
+					{
+						path.Add(current);
+						current = cameFrom[current];
+					}
+					path.Add(start);
+					path.Reverse();
+					return path;
+				}
+
+				openSet.Remove(current);
+				foreach (var neighbor in GetNeighbors(terrain, current, checkForDoors))
+				{
+					float tentativeGScore = gScore[current] + 1; // Assuming uniform cost
+					if (!gScore.ContainsKey(neighbor) || tentativeGScore < gScore[neighbor])
+					{
+						cameFrom[neighbor] = current;
+						gScore[neighbor] = tentativeGScore;
+						fScore[neighbor] = gScore[neighbor] + Heuristic(neighbor, end);
+						openSet.Add(neighbor);
+						lastValidPosition = neighbor;
+					}
+				}
+			}
+
+			if (allowIncompletePath)
+			{
+				while (lastValidPosition != start)
+				{
+					path.Add(lastValidPosition);
+					lastValidPosition = cameFrom[lastValidPosition];
+				}
+				path.Add(start);
+				path.Reverse();
+				return path;
+			}
+
+			return null; // No path found
+		}
+
+		private float Heuristic(Vector3Int a, Vector3Int b)
+		{
+			return Mathf.Abs(a.x - b.x) + Mathf.Abs(a.y - b.y); // Manhattan distance
+		}
+
+		private class FScoreComparer : IComparer<Vector3Int>
+		{
+			private readonly Dictionary<Vector3Int, float> fScore;
+
+			public FScoreComparer(Dictionary<Vector3Int, float> fScore)
+			{
+				this.fScore = fScore;
+			}
+
+			public int Compare(Vector3Int a, Vector3Int b)
+			{
+				if (fScore[a] < fScore[b]) return -1;
+				if (fScore[a] > fScore[b]) return 1;
+				return 0;
+			}
+		}
+
+		private IEnumerable<Vector3Int> GetNeighbors(ChunkedTileMap<MetaDataNode> terrain, Vector3Int position, bool checkForDoors)
+		{
+			var directions = new List<Vector3Int>
+			{
+				new(1, 0, 0),
+				new(-1, 0, 0),
+				new(0, 1, 0),
+				new(0, -1, 0)
+			};
+
+			foreach (var direction in directions)
+			{
+				var neighbor = position + direction;
+				if (PositionIsOutOfBounds(terrain, neighbor)) continue;
+				var tile = terrain.GetTile(neighbor);
+				if (tile == null) continue; // No tile at this position. Probably space.
+				if (tile is { IsOccupied: false })
+				{
+					yield return neighbor;
+				}
+				else
+				{
+					//(Max): This is a performance killer.
+					//We need to find a way to update tiles to register if their occupied type is made by something like a door/window/etc.
```

Comment:
There is a RegisterDoor class that doors specifically use, if you can access the register tile you could possibly do something like:

if(tile is RegisterDoor)
Though not sure if accessing a registertile on a given layer is easier or harder than checking for a door